extensions/amp-animation/0.1/web-animations.js

```diff
@@ -502,31 +624,46 @@ export class MeasureScanner extends Scanner {
     if (SERVICE_PROPS[prop]) {
       return;
     }
-    if (this.validate_) {
-      user().assert(isWhitelistedProp(prop),
-          'Property is not whitelisted for animation: %s', prop);
-    } else {
-      dev().assert(isWhitelistedProp(prop),
-          'Property is not whitelisted for animation: %s', prop);
-    }
+    user().assert(isWhitelistedProp(prop),
+        'Property is not whitelisted for animation: %s', prop);
   }
 
   /**
-   * Runs the callback with the specified timing and restores old timing
-   * later.
-   * @param {!WebAnimationTimingDef} timing
+   * Resolves common parameters of an animation: target, timing and vars.
+   * @param {!WebAnimationDef} spec
    * @param {function()} callback
    * @private
    */
-  withTiming_(timing, callback) {
+  with_(spec, callback) {
+    // Save context.
+    const prevTarget = this.target_;
+    const prevVars = this.vars_;
     const prevTiming = this.timing_;
-    this.timing_ = timing;
-    callback();
+
+    // Push new context and perform calculations.
+    const targets =
+        (spec.target || spec.selector) ?
+        this.resolveTargets_(spec) :
+        [null];
+    targets.forEach(target => {
+      this.target_ = target || prevTarget;
```

Comment:
`prevTarget` is basically the "parent" target. Either a nested animation overrides it or the "parent" target is used. The `mergeVars_` and `mergeTiming_` that follow, essentially do the same thing.